sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -214,10 +218,15 @@ private[sql] class JacksonGenerator(
   private def writeMapData(
       map: MapData, mapType: MapType, fieldWriter: ValueWriter): Unit = {
     val keyArray = map.keyArray()
+    val keyString = mapType.keyType match {
+      case CalendarIntervalType =>
```

Comment:
This can't happen actually. We don't allow writing out interval values. Do you have an example that can hit this code path?

Reply:
https://github.com/apache/spark/pull/26418/commits/3aa723ca6a811898ea4f8c2d6100c7e4045b0c30#diff-8a461909033e3f735f8239825970c378R218-L229

Reply:
ah `to_json` do not have the interval type check. makes sense.

Reply:
@yaooqinn, how about `to_csv`?